ansible/roles/shd-client-core/tasks/main.yml

```
@@ -3,6 +3,8 @@
     kubeconfig: ~/.kube/config
     namespace: "{{ NAMESPACE }}"
     template: svc.yml.j2
+    state: absent
+  when: WITH_SHD_CLIENT is not defined or not WITH_SHD_CLIENT | bool
   tags:
```

This is the solution for it
```suggestion
    template: svc.yml.j2
    state: "{{ 'present' if WITH_SHD_CLIENT else 'absent'}}"
  tags:
```